client/src/holdings/AddDividendDialog.tsx

```diff
@@ -95,11 +104,17 @@ export function AddDividendDialog({holding, onDividendCreated}: IProps) {
                 </Button>
                 <Button type='submit' disabled={isSubmitting} color='primary'>
                   Add
-                </Button>
+                </Button>{' '}
               </DialogActions>
             </Form>
           )}
         </Formik>
+        <Snackbar
```

Comment:
This way means we have the snackbar duplicated every time we want to use it. Instead I suggest:

1. Remove the snackbar HTML from everywhere except the HoldingsList
2. Make a function in HoldingsList that takes a message and opens the snackbar
3. Pass this function as a prop into AddDividendDialog, AddHoldingForm and AddTradeDialog, and then call it when you want to show an error.

An example of passing a function into a component is like how we pass the onTradeCreated function into AddTradeDialog

Reply:
@willbeaufoy function setup for delete holding currently only. Not able to work out how to pass the message as Props.

Reply:
I thought it would be easier to show you what I have in mind, so see this branch I've just done as a demo: https://github.com/willbeaufoy/portfolio-tracker/tree/snackbar-demo

We have the snackbar itself in App.tsx because it's independent of other UI components, then we pass down the function to open it into any components that need it.

Reply:
@willbeaufoy just added another Commit. I've added the severity in so now we can call the show notification function with the message and the type of notification we want as per the list here:

https://material-ui.com/components/snackbars/#customized-snackbars

It passes the severity state to the alert function which then displays the message. Also if you can, can you review the Refresh prices async function with is now try/catch. Console logs the error but won't display the notification.

Lastly once you are happy, we'll need to decide which API calls require a notification - delete/add/refresh etc